File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // first check if both string length not eqal then not anagram return false
        if(s.length() != t.length()){
            return false;
        }

        // crate charter array to mange count
        int charAnagram[] = new int[26]; // only have 26 alaphabets

        for(int i=0; i < s.length(); i++){
            charAnagram[s.charAt(i) - 'a'] ++;
            charAnagram[t.charAt(i) - 'a'] --;
        }

        for(int c: charAnagram){
            if(c != 0){
                return false;
            }
        }

        return true;
    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        countS, countT = {}, {}

        for row in range(len(s)):
            countS[s[row]] = 1 + countS.get(s[row], 0)
            countT[t[row]] = 1 + countT.get(t[row], 0)

        return countS == countT